### PR TITLE
VL-165_Remove-icon-slots_Dmytro-Holdobin

### PR DESCRIPTION
**Pull request by Explicit12**
https://ilevel.atlassian.net/browse/VL-165

### CODE REVIEW

File: src/ui.form-date-picker-range/UDatePickerRange.vue [hunk 1]
```diff
@@ -600,31 +600,25 @@ watchEffect(() => {
...
-        <!-- @slot Use it to add something before the date. -->
-        <slot name="left" />
-      </template>
-
-      <template #left-icon>
-        <!-- @slot Use it to add icon before the date. -->
-        <slot name="left-icon" />
+      <template #left="{ iconName, iconSize }">
+        <!--
+          @slot Use it to something before the date.
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/UDatePickerRange.vue:605]: @slot Use it to add something before the date.

File: src/ui.form-date-picker-range/UDatePickerRange.vue [hunk 2]
```diff
@@ -600,31 +600,25 @@ watchEffect(() => {
...
+          @binding {string} icon-size
+        -->
+        <slot name="left" :icon-name="iconName" :icon-size="iconSize" />
       </template>
 
-      <template #right-icon="{ iconName, iconSize }">
+      <template #right="{ iconName, iconSize }">
         <!--
-          @slot Use it add an icon after the date.
+          @slot Use it something after the date.
```
Comment (itsJohnnyGrid) [inline src/ui.form-date-picker-range/UDatePickerRange.vue:614]: @slot Use it to add something after the date.

File: src/ui.form-input-money/UInputMoney.vue [hunk 3]
```diff
@@ -134,16 +134,6 @@ const { moneyInputAttrs } = useUI<Config>(defaultConfig);
       <slot name="left" />
```
Comment (itsJohnnyGrid) [inline src/ui.form-input-money/UInputMoney.vue:134]: what about iсon size and name? (same as in datepicker)

File: src/ui.form-input-search/UInputSearch.vue [hunk 4]
```diff
@@ -159,7 +159,7 @@ const { config, searchInputAttrs, searchIconAttrs, clearIconAttrs, searchButtonA
       <slot name="left-icon" />
```
Comment (itsJohnnyGrid) [inline src/ui.form-input-search/UInputSearch.vue:159]: why not removed?

File: src/ui.form-input-search/UInputSearch.vue [hunk 5]
```diff
@@ -174,7 +174,7 @@ const { config, searchInputAttrs, searchIconAttrs, clearIconAttrs, searchButtonA
 
       <!-- @slot Use it to add icon after the text. -->
```
Comment (itsJohnnyGrid) [inline src/ui.form-input-search/UInputSearch.vue:175]: Use it to add something after the text.
icon name and size are missing

File: src/ui.form-input-search/config.ts [hunk 6]
```diff
@@ -2,7 +2,7 @@ export default /*tw*/ {
   searchInput: "{UInput}",
   clearIcon: "{UIcon}",
   searchIcon: "{UIcon}",
-  searchButton: "{UButton} rounded-l-none ml-1 outline outline-1 outline-gray-900",
+  searchButton: "{UButton} -mr-3 rounded-l-none ml-1 outline outline-1 outline-gray-900",
```
Comment (itsJohnnyGrid) [inline src/ui.form-input-search/config.ts:5]: no, you need to add mutatedProp and cva variant

Comment (Explicit12) [reply src/ui.form-input-search/config.ts:5]: Cva variant for what?

File: src/ui.form-input-money/UInputMoney.vue (hunk 3 above)
Comment (Explicit12) [reply src/ui.form-input-money/UInputMoney.vue]: I pulled main branch and looks like we don't have iconSize at all now (it set in config).

File: src/ui.form-input-search/config.ts (hunk 6 above)
Comment (itsJohnnyGrid) [reply src/ui.form-input-search/config.ts]: `-mr-3` is bad solution. You need to remove right slot padding, if search button label is not empty

File: src/ui.form-input-search/UInputSearch.vue [hunk 7]
```diff
@@ -151,10 +146,14 @@ const { config, searchInputAttrs, searchIconAttrs, clearIconAttrs, searchButtonA
         @click="onClickClear"
       />
 
-      <!-- @slot Use it to add icon after the text. -->
+      <!-- 
+        @slot Use it to add something after the text. 
+        @binding {string} icon-name
+        @binding {string} icon-size
```
Comment (itsJohnnyGrid) [inline src/ui.form-input-search/UInputSearch.vue:152]: `@binding {string} icon-size` but there is no `icon-size` slot prop. remove it at all.

File: src/ui.form-input-money/UInputMoney.vue [hunk 8]
```diff
@@ -130,13 +130,21 @@ const { moneyInputAttrs } = useUI<Config>(defaultConfig);
...
 
     <template #right>
-      <!-- @slot Use it to add something right. -->
-      <slot name="right" />
+      <!-- 
+        @slot Use it to add something right. 
+        @binding {string} icon-name
+        @binding {string} icon-size
+      -->
+      <slot name="right" :icon-name="leftIcon" :icon-size="size" />
```
Comment (itsJohnnyGrid) [inline src/ui.form-input-money/UInputMoney.vue:147]: so, remove `:icon-size="size"` slot prop from all slots at all. And related slot docs.

**Review by itsJohnnyGrid — APPROVED** [review]

**Pull request merged**